models/pose_estimation_mediapipe/demo.cpp

```diff
@@ -0,0 +1,2913 @@
+        Mat bbox = (Mat_<float>(2, 2) << vmin0, vmin1, vmax0, vmax1);
+        Mat centerBox;
+        reduce(bbox, centerBox, 0, REDUCE_AVG, CV_32F);
+        Mat whBox = bbox.row(1) - bbox.row(0);
+        Mat newHalfSize = whBox * this->personBoxEnlargeFactor / 2;
+        vector<Mat> vmat(2);
+        vmat[0] = centerBox - newHalfSize;
+        vmat[1] = centerBox + newHalfSize;
+        vconcat(vmat, bbox);
+        // invert rotation for mask
+        mask = mask.reshape(1, 256);
+        Mat invertRotationMatrix = getRotationMatrix2D(Point(mask.cols / 2, mask.rows / 2), -angle, 1.0);
+        Mat invertRotationMask;
+        warpAffine(mask, invertRotationMask, invertRotationMatrix, Size(mask.cols, mask.rows));
+        // enlarge mask
+        resize(invertRotationMask, invertRotationMask, Size(whRotatedPersonPbox.at<float>(0), whRotatedPersonPbox.at<float>(1)));
+        // crop and pad mask
+        int minW = -min(padBias.width, 0);
+        int minH= -min(padBias.height, 0);
+        int left = max(padBias.width, 0);
+        int top = max(padBias.height, 0);
+        Size padOver = imgSize - Size(invertRotationMask.cols, invertRotationMask.rows) - padBias;
+        int maxW = min(padOver.width, 0) + invertRotationMask.cols;
+        int maxH = min(padOver.height, 0) + invertRotationMask.rows;
+        int right = max(padOver.width, 0);
+        int bottom = max(padOver.height, 0);
+        invertRotationMask = invertRotationMask(Rect(minW, minH, maxW - minW, maxH - minH)).clone();
+        copyMakeBorder(invertRotationMask, invertRotationMask, top, bottom, left, right, BORDER_CONSTANT, Scalar::all(0));
+        // binarize mask
+        threshold(invertRotationMask, invertRotationMask, 1, 255, THRESH_BINARY);
+
+        /* 2*2 person bbox: [[x1, y1], [x2, y2]]
+        # 39*5 screen landmarks: 33 keypoints and 6 auxiliary points with [x, y, z, visibility, presence], z value is relative to HIP
+        # Visibility is probability that a keypoint is located within the frame and not occluded by another bigger body part or another object
+        # Presence is probability that a keypoint is located within the frame
+        # 39*3 world landmarks: 33 keypoints and 6 auxiliary points with [x, y, z] 3D metric x, y, z coordinate
+        # img_height*img_width mask: gray mask, where 255 indicates the full body of a person and 0 means background
+        # 64*64*39 heatmap: currently only used for refining landmarks, requires sigmod processing before use
+        # conf: confidence of prediction*/
+        return tuple<Mat , Mat, Mat, Mat, Mat, float>(bbox, landmarks, rotatedLandmarksWorld, invertRotationMask, heatmap, valConf);
+    }
+};
+
+std::string keys =
+"{ help  h          |                                               | Print help message. }"
+"{ model m          | person_detection_mediapipe_2023mar.onnx       | Usage: Path to the model, defaults to person_detection_mediapipe_2023mar.onnx  }"
```

Comment:
The model should be `pose_estimation_mediapipe_2023mar.onnx`. Since this task use 2 models, should we make 2 arguments for detection model path and estimation model path? How do you think @fengyuentau ?

Reply:
wiating for answer @fengyuentau

Reply:
My fault, we can just use one model path, and the detection model path we can just use `../person_detection_mediapipe/xxxx.onnx`